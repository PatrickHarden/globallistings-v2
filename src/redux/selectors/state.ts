import { State } from '../../types/state';
import { theme } from '../../themes/default/theme';
import { createSelector } from 'reselect';
import { FilterType } from '../../types/config';

// State
export const entityMetaSelector = (state: State) => state.entitiesMeta;
export const entitiesSelector = (state: State) => state.entities;
export const uiSelector = (state: State) => state.ui;
export const locationSelector = (state: State) => state.router.location;

// Entities
export const propertiesSelector = createSelector(
  entitiesSelector,
  entities => entities.properties
);

export const geocodeSelector = createSelector(
  entitiesSelector,
  entities => entities.geocode
);

export const propertySearchesSelector = createSelector(
  entitiesSelector,
  entities => entities.propertySearches
);

export const configSelector = createSelector(
  entitiesSelector,
  entities => entities.config
);

export const autocompleteSelector = createSelector(
  entitiesSelector,
  entities => entities.autocomplete
);

export const autocompleteSearchesSelector = createSelector(
  entitiesSelector,
  entities => entities.autocompleteSearches
);

// Meta
export const propertiesMetaSelector = createSelector(
  entityMetaSelector,
  meta => meta.propertySearches
);

export const geocodeMetaSelector = createSelector(
  entityMetaSelector,
  meta => meta.geocode
);

export const configMetaSelector = createSelector(
  entityMetaSelector,
  meta => meta.config
);

export const autocompleteSearchesMetaSelector = createSelector(
  entityMetaSelector,
  meta => meta.autocompleteSearches
);

// Config
export const localeSelector = createSelector(
  configSelector,
  config => config.language
);

export const messagesSelector = createSelector(
  configSelector,
  config => config.i18n
);

export const filtersSelector = createSelector(
  configSelector,
  config => config.filters
);

export const rangeFiltersSelector = createSelector(
  filtersSelector,
  filters => filters.filter(f => f.type === FilterType.range)
);

// Other

// TODO add logic to switch between themes based on config
export const themeSelector = (state: State) => theme;

// TODO figure out where it will be coming from
export const googleMapsApiKeySelector = () => 'AIzaSyCzEw2NaQFixR7sZCLnZ1oGQljW7AxLZ7k';
