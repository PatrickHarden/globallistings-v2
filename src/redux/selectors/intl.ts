import { createSelector } from 'reselect';
import { IntlProvider } from 'react-intl';
import { Dictionary } from '../../types/common/dictionary';
import { localeSelector, messagesSelector } from './state';

export const intlSelector = createSelector(
  localeSelector,
  messagesSelector,
  (locale: string, messages: Dictionary<string>) =>
    new IntlProvider({ locale, messages }).getChildContext().intl
);
