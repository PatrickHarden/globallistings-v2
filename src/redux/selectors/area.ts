import { createSelector } from 'reselect';

import { Bounds, LatLng } from '../../types/state';
import { DistanceUnit } from '../../types/config';

import { googleBoundsToBounds } from '../../utils/bounds';
import { haversineRadius } from '../../utils/haversine';
import { paramsSelector } from './config/config';
import { defaultFiltersParamsSelectors } from './config/filters';

import { plpLocationRadiusSelector, plpLocationSelector, locationParamsSelector } from './location';

interface BaseArea {
  readonly bounds: Bounds;
  readonly center: LatLng;
}

export interface RectangleArea extends BaseArea {
  readonly type: 'rectangle';
}

export interface CircleArea extends BaseArea {
  readonly radius: number;
  readonly type: 'circle';
}

export type Area = RectangleArea | CircleArea;

const defaultRadiusSelector = createSelector(
  defaultFiltersParamsSelectors,
  ({ radius }) => {
    if (Array.isArray(radius)) {
      throw new Error('radius can not be an array');
    }

    return radius;
  }
);

const radiusSelector = createSelector(
  defaultRadiusSelector,
  plpLocationRadiusSelector,
  (defaultRadius, radius) => (radius ? radius : defaultRadius)
);

const radiusTypeSelector = createSelector(
  paramsSelector,
  params => (params && params.RadiusType === 'Kilometers' ? DistanceUnit.km : DistanceUnit.miles)
);

export const areaSelector = createSelector(
  radiusSelector,
  plpLocationSelector,
  radiusTypeSelector,
  (radius, location, radiusType): Area => {
    if (!radius || radius === '' || radius === '0') {
      return {
        type: 'rectangle',
        bounds: location.bounds,
        center: location.center,
      };
    }

    const resultRadius = haversineRadius(
      location.bounds,
      location.center,
      Number.parseFloat(radius),
      radiusType
    );

    const boundsCircle = new google.maps.Circle({
      center: location.center,
      radius: resultRadius * 1000,
    });

    return {
      type: 'circle',
      radius: resultRadius,
      bounds: googleBoundsToBounds(boundsCircle.getBounds()),
      center: location.center,
    };
  }
);
