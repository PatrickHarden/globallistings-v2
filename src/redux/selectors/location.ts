import { createSelector } from 'reselect';
import { locationSelector, geocodeSelector } from './state';
import { parseUrl } from 'query-string';
import { plpUrlPathParamsSelector, pathParamsSelector } from './match';

export const queryStringSelector = createSelector(
  locationSelector,
  location => location.search
);

export const pathSelector = createSelector(
  locationSelector,
  location => location.pathname
);

export const locationQuerySelector = createSelector(
  queryStringSelector,
  queryString => parseUrl(queryString).query
);

export const querySelector = createSelector(
  queryStringSelector,
  location => parseUrl(location).query
);

export const locationParamsSelector = createSelector(
  locationQuerySelector,
  pathParamsSelector,
  (query, params) => ({ ...query, ...params })
);

export const plpLocationNameSelector = createSelector(
  plpUrlPathParamsSelector,
  params => params.location
);

export const plpTransactionTypeSelector = createSelector(
  plpUrlPathParamsSelector,
  params => params.transactionType
);

export const plpLocationSelector = createSelector(
  plpLocationNameSelector,
  geocodeSelector,
  (locationName, geocode) => {
    const location = geocode[locationName];
    if (!location) {
      throw new Error('Failed retrieving geocoding for the location');
    }

    return location;
  }
);

export const plpLocationRadiusSelector = createSelector(
  locationQuerySelector,
  ({ radius }) => {
    if (Array.isArray(radius)) {
      throw new Error('radius can not be an array');
    }

    return radius;
  }
);
