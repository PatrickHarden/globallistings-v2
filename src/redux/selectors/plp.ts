import { createSelector } from 'reselect';
import {
  geocodeMetaSelector,
  propertiesMetaSelector,
  propertySearchesSelector,
  configSelector,
} from './state';
import { propertiesSearchRequestQuerySelector } from './search-api-request';
import { murmur2 } from 'murmurhash-js';
import { plpLocationNameSelector } from './location';
import { otherConfigsSelector } from './config/config';
import { plpUrlPathParamsSelector, plpMatchSelector } from './match';

export const plpQueryHashSelector = createSelector(
  propertiesSearchRequestQuerySelector,
  query => murmur2(JSON.stringify(query)).toString()
);

export const plpPropertyRequestStatus = createSelector(
  plpQueryHashSelector,
  propertiesMetaSelector,
  (hash, propertiesMeta) => {
    const propertyMeta = propertiesMeta[hash];
    if (propertyMeta === undefined) {
      return 'NONE';
    }

    return propertyMeta.status;
  }
);

export const propertySearchSelector = createSelector(
  propertySearchesSelector,
  plpQueryHashSelector,
  (propertySearches, hash) => {
    const result = propertySearches[hash];

    if (result) {
      return result;
    }

    return [];
  }
);

const plpLocationGeocodeStatus = createSelector(
  plpLocationNameSelector,
  geocodeMetaSelector,
  (location, meta) => {
    const entityMeta = meta[location];
    return entityMeta ? entityMeta.status : 'NONE';
  }
);

export const isPlpLocationGeocoded = createSelector(
  plpLocationGeocodeStatus,
  status => status === 'SUCCESS'
);

export const isUsageTypeConfigLoaded = createSelector(
  configSelector,
  plpUrlPathParamsSelector,
  (config, path) => {
    if (!config.currentConfig) {
      return true;
    }

    return path.usageType === config.currentConfig;
  }
);

export const usageTypeSelector = createSelector(
  plpUrlPathParamsSelector,
  params => params.usageType
);

export const configEndpointForCurrentUsageTypeSelector = createSelector(
  usageTypeSelector,
  otherConfigsSelector,
  (usageType, configs) => {
    const config = configs.find(c => c.name === usageType);
    if (!config) {
      throw new Error('Config is missing data for config endpoint mapping');
    }

    return config.endpoint;
  }
);

export const isPlpPath = createSelector(
  plpMatchSelector,
  match => !!match
);
