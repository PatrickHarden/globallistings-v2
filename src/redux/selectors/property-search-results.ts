import { createSelector } from 'reselect';
import { InjectedIntl } from 'react-intl';
import { propertiesSelector } from './state';
import { intlSelector } from './intl';
import { notEmpty } from '../../types/common/notEmpty';
import { Location, Property } from '../../types/property';
import { propertySearchSelector, usageTypeSelector } from './plp';
import { selectedPropertyPinSelector, highlightedPropertySelector } from './ui';
import { plpTransactionTypeSelector } from './location';

import { mapAddress } from '../../utils/propertyData/mapAddress';
import { mapImage } from '../../utils/propertyData/mapImage';
import { mapPrice } from '../../utils/propertyData/mapPrice';
import { mapSize } from '../../utils/propertyData/mapSize';
import { capitalise } from '../../utils/capitalise';

export interface PropertyOverview {
  readonly imgUrl: string;
  readonly price?: string;
  readonly size?: string;
  readonly address: string;
  readonly location: Location;
  readonly id: string;
  readonly selected: boolean;
}

// TODO update to a real one
export const isCommercialSelector = () => true;

const toPropertyOverview = (
  intl: InjectedIntl,
  isCommercial: boolean,
  selectedPropertyId: string,
  highlightedPropertyId: string
) => (p: Property, i: Number): PropertyOverview => ({
  address: mapAddress(p, intl),
  imgUrl: mapImage(p),
  price: mapPrice(p, intl),
  size: mapSize(p, intl, isCommercial),
  location: p.location,
  id: p.id,
  // Selected poperty = currently highlighted (mouseover list), selected (pin clicked) or first in array
  selected: highlightedPropertyId
    ? highlightedPropertyId === p.id
    : selectedPropertyId === p.id || i === 0,
});

export const propertiesSearchResultsSelector = createSelector(
  propertiesSelector,
  propertySearchSelector,
  intlSelector,
  isCommercialSelector,
  selectedPropertyPinSelector,
  highlightedPropertySelector,
  (properties, propertySearch, intl, isCommercial, selectedPropertyId, highlightedPropertyId) => {
    return propertySearch
      .map(p => properties[p])
      .filter(notEmpty)
      .map(toPropertyOverview(intl, isCommercial, selectedPropertyId, highlightedPropertyId));
  }
);

export const resultCountSelector = createSelector(
  propertiesSearchResultsSelector,
  usageTypeSelector,
  plpTransactionTypeSelector,
  intlSelector,
  (results, usage, transaction, { formatMessage }) =>
    formatMessage(
      { id: 'PropertiesFound' },
      {
        propertyCount: results.length,
        propertyTypePlural: formatMessage({
          id: `PDPPluralPropertyType${capitalise(usage)}`,
        }).toLocaleLowerCase(),
        searchType: formatMessage({
          id: transaction === 'sale' ? 'saleSearchType' : 'letSearchType',
        }),
      }
    )
);
