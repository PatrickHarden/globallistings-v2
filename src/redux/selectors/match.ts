import { createMatchSelector, RouterRootState } from 'connected-react-router';
import { PlpPath } from '../../types/plp';
import { PdpPath } from '../../types/pdp';
import RoutePaths from '../../constants/routePaths';
import { createSelector } from 'reselect';

export const plpMatchSelector = createMatchSelector<RouterRootState, PlpPath>(RoutePaths.plp);
export const pdpMatchSelector = createMatchSelector<RouterRootState, PdpPath>(RoutePaths.pdp);

export const pathParamsSelector = createSelector(
  plpMatchSelector,
  pdpMatchSelector,
  (plpMatch, pdpMatch) => ({ ...(plpMatch || {}), ...(pdpMatch || {}) })
);

export const plpUrlPathParamsSelector = createSelector(
  plpMatchSelector,
  match => {
    if (!match) {
      throw new Error('Url path must be defined to use this selector');
    }
    return match.params;
  }
);

export const pdpUrlPathParamsSelector = createSelector(
  pdpMatchSelector,
  match => {
    if (!match) {
      throw new Error('Url path must be defined to use this selector');
    }
    return match.params;
  }
);
