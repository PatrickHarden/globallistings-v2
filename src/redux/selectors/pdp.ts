import { createSelector } from 'reselect';
import { InjectedIntl } from 'react-intl';

import { pdpUrlPathParamsSelector, pdpMatchSelector } from './match';
import { propertiesSelector } from './state';
import { intlSelector } from './intl';
import { Location, Property, Description } from '../../types/property';

import { mapAddress } from '../../utils/propertyData/mapAddress';
import { mapImage } from '../../utils/propertyData/mapImage';
import { mapPrice } from '../../utils/propertyData/mapPrice';
import { mapSize } from '../../utils/propertyData/mapSize';
import { mapSpecification } from '../../utils/propertyData/mapSpecification';

import { isCommercialSelector } from './property-search-results';

export interface PropertyDetails {
  readonly address: string;
  readonly id: string;
  readonly imgUrl: string;
  readonly location: Location;
  readonly description?: Description;
  readonly price?: string;
  readonly size?: string;
  readonly specification?: string[];
}

const toPropertyDetails = (
  p: Property,
  intl: InjectedIntl,
  isCommercial: boolean
): PropertyDetails => ({
  address: mapAddress(p, intl, 'AddressSummaryLong'),
  description: p.description,
  id: p.id,
  imgUrl: mapImage(p),
  location: p.location,
  price: mapPrice(p, intl),
  size: mapSize(p, intl, isCommercial),
  specification: mapSpecification(p, intl),
});

export const propertyIdSelector = createSelector(
  pdpUrlPathParamsSelector,
  params => params.propertyId
);

export const propertySelector = createSelector(
  propertiesSelector,
  propertyIdSelector,
  (properties, id) => properties[id]
);

export const propertyDetailsSelector = createSelector(
  propertySelector,
  intlSelector,
  isCommercialSelector,
  (property, intl, isCommercial) => property && toPropertyDetails(property, intl, isCommercial)
);

export const sectionHeadingsSelector = createSelector(
  intlSelector,
  intl => ({
    details: intl.formatMessage({ id: 'OverviewTitle' }),
    specification: intl.formatMessage({ id: 'PdpSpecificationTitle' }),
  })
);

export const isPdpPath = createSelector(
  pdpMatchSelector,
  match => !!match
);
