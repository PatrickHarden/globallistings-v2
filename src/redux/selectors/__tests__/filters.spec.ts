jest.mock('../../selectors/match');

import {
  filtersInfoSelector,
  RangeFilterInfo,
  defaultFiltersParamsSelectors,
} from '../config/filters';
import { DeepPartial } from 'redux';
import { State } from '../../../types/state';

import { pathParamsSelector } from '../../selectors/match';
((pathParamsSelector as unknown) as jest.Mock).mockReturnValue({});

describe(filtersInfoSelector, () => {
  it('should correctly map radius param to an option', () => {
    const state: DeepPartial<State> = {
      router: { location: { search: '?radius=1' } },
      entities: {
        config: {
          filters: [
            {
              name: 'radius',
              type: 'select',
              options: [
                { value: '', label: 'None' },
                { value: '1', label: 'One' },
                { value: '2', label: 'Two' },
              ],
            },
          ],
        },
      },
      ui: { search: {} },
    };

    expect(filtersInfoSelector(state as State)).toMatchSnapshot();
  });

  it('should correctly map range param to an option', () => {
    const state: DeepPartial<State> = {
      router: { location: { search: '?minValue|maxValue_min=1&minValue|maxValue_max=2' } },
      entities: {
        config: {
          filters: [
            {
              name: 'minValue|maxValue',
              type: 'range',
              minValues: [
                { value: '', label: 'None', type: 'min' },
                { value: '1', label: 'One', type: 'min' },
                { value: '2', label: 'Two', type: 'min' },
              ],
              maxValues: [
                { value: '', label: 'None', type: 'max' },
                { value: '3', label: 'Three', type: 'max' },
                { value: '4', label: 'Four', type: 'max' },
              ],
            },
          ],
        },
      },
      ui: { search: {} },
    };

    expect(filtersInfoSelector(state as State)).toMatchSnapshot();
  });

  it('should remove invalid options from a range filter', () => {
    const state: DeepPartial<State> = {
      router: { location: { search: '?minValue|maxValue_min=2&minValue|maxValue_max=3' } },
      entities: {
        config: {
          filters: [
            {
              name: 'minValue|maxValue',
              type: 'range',
              minValues: [
                { value: '', label: 'None', type: 'min' },
                { value: '1', label: 'One', type: 'min' },
                { value: '2', label: 'Two', type: 'min' },
                { value: '3', label: 'Three', type: 'min' },
                { value: '4', label: 'four', type: 'min' },
              ],
              maxValues: [
                { value: '', label: 'None', type: 'max' },
                { value: '1', label: 'One', type: 'max' },
                { value: '2', label: 'Two', type: 'max' },
                { value: '3', label: 'Three', type: 'max' },
                { value: '4', label: 'four', type: 'max' },
              ],
            },
          ],
        },
      },
      ui: { search: {} },
    };

    expect((filtersInfoSelector(state as State)[0] as RangeFilterInfo).min.items).toEqual([
      { value: '', label: 'None', type: 'min' },
      { value: '1', label: 'One', type: 'min' },
      { value: '2', label: 'Two', type: 'min' },
      { value: '3', label: 'Three', type: 'min' },
    ]);

    expect((filtersInfoSelector(state as State)[0] as RangeFilterInfo).max.items).toEqual([
      { value: '', label: 'None', type: 'max' },
      { value: '2', label: 'Two', type: 'max' },
      { value: '3', label: 'Three', type: 'max' },
      { value: '4', label: 'four', type: 'max' },
    ]);
  });

  it('should correctly map radius param', () => {
    const state: DeepPartial<State> = {
      router: { location: { search: '?radius[]=1&radius[]=2' } },
      entities: {
        config: {
          filters: [
            {
              name: 'radius',
              type: 'select',
              options: [
                { value: '', label: 'None' },
                { value: '1', label: 'One' },
                { value: '2', label: 'Two' },
              ],
            },
          ],
        },
      },
      ui: { search: {} },
    };

    expect(filtersInfoSelector(state as State)).toMatchSnapshot();
  });

  it('should handle unmatched query correctly', () => {
    const state: DeepPartial<State> = {
      router: { location: { search: '?radius=3' } },
      entities: {
        config: {
          filters: [
            {
              name: 'radius',
              type: 'select',
              options: [
                { value: '', label: 'None' },
                { value: '1', label: 'One' },
                { value: '2', label: 'Two' },
              ],
            },
          ],
        },
      },
      ui: { search: {} },
    };

    expect(filtersInfoSelector(state as State)).toMatchSnapshot();
  });
});

describe(defaultFiltersParamsSelectors, () => {
  it('with no default options should return empty object', () => {
    const state: DeepPartial<State> = {
      entities: {
        config: {
          filters: [
            {
              name: 'radius',
              type: 'select',
              options: [
                { value: '', label: 'None' },
                { value: '1', label: 'One' },
                { value: '2', label: 'Two' },
              ],
            },
          ],
        },
      },
      ui: { search: {} },
    };

    const result = defaultFiltersParamsSelectors(state as State);

    expect(result).toEqual({});
  });

  it('with a default option should return correct params', () => {
    const state: DeepPartial<State> = {
      entities: {
        config: {
          filters: [
            {
              name: 'radius',
              type: 'select',
              options: [
                { value: '', label: 'None' },
                { value: '1', label: 'One' },
                { value: '2', label: 'Two', default: true },
              ],
            },
          ],
        },
      },
      ui: { search: {} },
    };

    const result = defaultFiltersParamsSelectors(state as State);

    expect(result).toEqual({ radius: '2' });
  });

  it('with multiple default options should return correct params', () => {
    const state: DeepPartial<State> = {
      entities: {
        config: {
          filters: [
            {
              name: 'radius',
              type: 'select',
              options: [
                { value: '', label: 'None' },
                { value: '1', label: 'One' },
                { value: '2', label: 'Two', default: true },
              ],
            },
            {
              name: 'area',
              type: 'select',
              options: [
                { value: 'A', label: 'Area A' },
                { value: 'B', label: 'Area B', default: true },
              ],
            },
          ],
        },
      },
      ui: { search: {} },
    };

    const result = defaultFiltersParamsSelectors(state as State);

    expect(result).toEqual({ radius: '2', area: 'B' });
  });

  it('with multiple default options should return correct params', () => {
    const state: DeepPartial<State> = {
      entities: {
        config: {
          filters: [],
        },
      },
      ui: { search: {} },
    };

    const result = defaultFiltersParamsSelectors(state as State);

    expect(result).toEqual({});
  });
});
