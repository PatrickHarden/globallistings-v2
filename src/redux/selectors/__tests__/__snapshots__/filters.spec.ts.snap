// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly map radius param 1`] = `
Array [
  Object {
    "items": Array [
      Object {
        "label": "None",
        "value": "",
      },
      Object {
        "label": "One",
        "value": "1",
      },
      Object {
        "label": "Two",
        "value": "2",
      },
    ],
    "name": "radius",
    "options": Array [
      Object {
        "label": "None",
        "value": "",
      },
      Object {
        "label": "One",
        "value": "1",
      },
      Object {
        "label": "Two",
        "value": "2",
      },
    ],
    "selectedItem": undefined,
    "type": "select",
  },
]
`;

exports[`should correctly map radius param to an option 1`] = `
Array [
  Object {
    "items": Array [
      Object {
        "label": "None",
        "value": "",
      },
      Object {
        "label": "One",
        "value": "1",
      },
      Object {
        "label": "Two",
        "value": "2",
      },
    ],
    "name": "radius",
    "options": Array [
      Object {
        "label": "None",
        "value": "",
      },
      Object {
        "label": "One",
        "value": "1",
      },
      Object {
        "label": "Two",
        "value": "2",
      },
    ],
    "selectedItem": Object {
      "label": "One",
      "value": "1",
    },
    "type": "select",
  },
]
`;

exports[`should correctly map range param to an option 1`] = `
Array [
  Object {
    "label": undefined,
    "max": Object {
      "items": Array [
        Object {
          "label": "None",
          "type": "max",
          "value": "",
        },
        Object {
          "label": "Three",
          "type": "max",
          "value": "3",
        },
        Object {
          "label": "Four",
          "type": "max",
          "value": "4",
        },
      ],
      "selectedItem": undefined,
    },
    "min": Object {
      "items": Array [
        Object {
          "label": "None",
          "type": "min",
          "value": "",
        },
        Object {
          "label": "One",
          "type": "min",
          "value": "1",
        },
        Object {
          "label": "Two",
          "type": "min",
          "value": "2",
        },
      ],
      "selectedItem": Object {
        "label": "One",
        "type": "min",
        "value": "1",
      },
    },
    "modifier": undefined,
    "name": "minValue|maxValue",
    "placement": undefined,
    "range": "range",
    "type": "range",
  },
]
`;

exports[`should handle unmatched query correctly 1`] = `
Array [
  Object {
    "items": Array [
      Object {
        "label": "None",
        "value": "",
      },
      Object {
        "label": "One",
        "value": "1",
      },
      Object {
        "label": "Two",
        "value": "2",
      },
    ],
    "name": "radius",
    "options": Array [
      Object {
        "label": "None",
        "value": "",
      },
      Object {
        "label": "One",
        "value": "1",
      },
      Object {
        "label": "Two",
        "value": "2",
      },
    ],
    "selectedItem": undefined,
    "type": "select",
  },
]
`;
